### PR TITLE
Backend authentication

### PR DESCRIPTION
**Pull request by domgud**
Added basic user seeding and roles (so you could authorize stuff :^) ). Also added restaurant registration request basic stuff (will need to come back and reimplement the confirmation with enums). Ples confirm my code 🤠

### CODE REVIEW

File: FoodApp/Controllers/RequestsController.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            return View(await _context.Restaurant.Where(r => r.State=="Pending") .ToListAsync());
+        }
+
+        public async Task<IActionResult> Details(string id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
```
Comment (RainbowDog98) [inline FoodApp/Controllers/RequestsController.cs:32]: We could use this instead for more readability.
`if (id is null) return NotFound();`

File: FoodApp/Data/ApplicationDbContext.cs [hunk 2]
```diff
@@ -14,5 +14,6 @@ public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
         {
         }
         public DbSet<FoodApp.Models.Dish> Dish { get; set; }
+        public DbSet<FoodApp.Models.Restaurant> Restaurant { get; set; }
```
Comment (RainbowDog98) [inline FoodApp/Data/ApplicationDbContext.cs:17]: Is it possible to use `DbSet<Dish/Restaurant>` instead? Seems unnecesserly long.

File: FoodApp/Models/Dish.cs [hunk 3]
```diff
@@ -8,9 +9,13 @@ namespace FoodApp.Models
     public class Dish
     {
         public int Id { get; set; }
+        [Required]
         public string Name { get; set; }
+        [Required]
         public string Details { get; set; }
         public double Price { get; set; }
```
Comment (RainbowDog98) [inline FoodApp/Models/Dish.cs:16]: is Price not required for a dish? Also there should some `[Range()]` param set too. Negative prices are not very common :^)

File: FoodApp/Startup.cs [hunk 4]
```diff
@@ -72,6 +73,88 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
+        {
+            Task<IdentityUser> testUser = userManager.FindByEmailAsync(email);
+            testUser.Wait();
+
+            if (testUser.Result == null)
+            {
+                
+                IdentityUser user = new IdentityUser();
+                user.Email = email;
+                user.UserName = email;
```
Comment (RainbowDog98) [inline FoodApp/Startup.cs:121]: Is this here correct? Seems awfully bold to just have the username the same as the email. May as well just use Email for identification.

File: FoodApp/Startup.cs [hunk 5]
```diff
@@ -72,6 +73,88 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
+        {
+            Task<IdentityUser> testUser = userManager.FindByEmailAsync(email);
+            testUser.Wait();
+
+            if (testUser.Result == null)
+            {
+
+                Restaurant user = new Restaurant();
+                user.Email = email;
+                user.UserName = email;
```
Comment (RainbowDog98) [inline FoodApp/Startup.cs:143]: The same question as in new User creation.

File: FoodApp/Controllers/RequestsController.cs (hunk 1 above)
Comment (domgud) [reply FoodApp/Controllers/RequestsController.cs:32]: No :)

File: FoodApp/Data/ApplicationDbContext.cs (hunk 2 above)
Comment (domgud) [reply FoodApp/Data/ApplicationDbContext.cs:17]: Easier to read :^)

File: FoodApp/Startup.cs (hunk 4 above)
Comment (domgud) [reply FoodApp/Startup.cs:121]: Not relevant. Mock data that isn't used wouldn't be used in production

File: FoodApp/Models/Dish.cs (hunk 3 above)
Comment (domgud) [reply FoodApp/Models/Dish.cs:16]: Will look into it more in next commits :)

File: FoodApp/Data/ApplicationDbContext.cs (hunk 2 above)
Comment (RainbowDog98) [reply FoodApp/Data/ApplicationDbContext.cs:17]: No it's not. It's unneeded repetition.

**Review by RainbowDog98 — APPROVED** [review]

**Pull request merged**